docs/ByteTree.md

```diff
@@ -0,0 +1,26 @@
+# ByteTree
+
+The ByteTree format is a binary format to efficiently serialize and deserialize trees. It was designed to serialize the syntax tree in `libSyntax` but the framework allows serialisation of arbitrary trees. It currently offers a serialiser written in C++ and a deserialiser written in Swift.
+
+## Overview
+The ByteTree format consists of two different constructs: *objects* and *scalars*. A scalar is a raw sequence of binary data. Scalars are untyped and the meaning of their binary data needs to be inferred by the client based on their position in the tree. An object consists of multiple *fields*, indexed by their position within the object, which again can be either objects or scalars. 
+
+## Serialization of scalars
+
+A scalar is encoded as its size followed by the data. Size is a `uint_32` that represents the size of the data in bytes in little endian order.
```

Comment:
Currently, it's implemented with  "native byte order".

Reply:
Ah, forgot to update the code on the deserializing side. Should be updated now.